eth2/types/src/slot_epoch.rs

```diff
@@ -38,7 +37,7 @@ impl Slot {
     }
 
     pub fn epoch(self, slots_per_epoch: u64) -> Epoch {
-        Epoch::from(self.0 / slots_per_epoch)
+        Epoch::from(self.0) / Epoch::from(slots_per_epoch)
```

Comment:
maybe use `checked_div`?

Reply:
Hmm yeah, I figured this was quite innocuous here, because `slots_per_epoch` is unlikely to ever be 0, and it would at least panic rather than wrapping in a strange way. I'll see how much flow-on effect this would have though.

Reply:
Perhaps we do an explicit `panic` if propagating the `Result` is too onerous (I suspect this will be the case). That way we get the panic in release too.

Reply:
I used `Epoch` division here because it will explicitly panic:

https://github.com/sigp/lighthouse/blob/3e4dc7ac4ee05fba2d47b6aaa2f6fbd65324d447/eth2/types/src/slot_epoch_macros.rs#L108-L114

Reply:
I'm going to leave this as-is, if that's OK with everyone.. I count 60 uses of `.epoch()` all throughout different crates, almost always with `T::slots_per_epoch()` as the argument, so I don't think it's worth it.

Reply:
Fine by me. Setting `SLOTS_PER_EPOCH` is not reasonable and there's plenty of other panics we could cause by setting "constants" to unreasonable values.

Reply:
It's also not really "user supplied input". I'd say it's "programmer supplied input".